docs/DeveloperGuide.md

```diff
@@ -1,34 +1,369 @@
+
+*Figure 7. Sequence diagram of setting the lastSeenList after a find command*
+
+<br>
+<br>
+
+### 3.4 Displaying the list after `find` or `list` commands
+
+#### 3.4.1 Current implementation
+The mechanism used to display the lastSeenList invoked by the list or find commands is facilitated by the listDrawer and findDrawer class respectively. They both work the same way but the list produced by findDrawer has an extra column which contains the dates of the respective entries.
+
+The following sequence diagram shows how the listDrawer class is used to display the lastSeenList.
+
+![list_Drawer](diagrams/listDrawer.png)
+
+*Figure 8. Sequence diagram of the usage of listDrawer to display the list*
+
+<br>
+<br>
+
+### 3.5 Edit activity in list feature
+
+#### 3.5.1 Current Implementation
+
+The editing mechanism is used by `EditFoodCommand` and `EditExerciseCommand` to amend the current list of activities.
+
+The following Sequence Diagram shows how `EditFoodCommand` is carried out when the user issues edit command, in this case, `edit 1 f/ egg c/ 10`:
+
+![Edit Activity](diagrams/EditActivityFeature.jpg)
+
+*Figure 9. Sequence diagram of edit food feature*
+
+> `EditExerciseCommand` diagram has a similar logic.
+
+<br>
+
+#### 3.5.2 Design Considerations
+
+Aspect: How to edit activity
+
+>Alternative 1 (current choice): Same command able to edit both activities, food and exercise in list.
+>* Pros: Able to guarantee that no abnormal behaviour will happen.
+>* Cons: Have to ensure that the different type of editing is implemented correctly.
+
+>Alternative 2: Have separate commands for editing the different activity type.
+>* Pros: Clear distinction of the classes.
+>* Cons: Increase in number of lines. Separate methods with similar logic will be created.
+
+<br>
+<br>
+
+### 3.6 Chaining feature
+
+#### 3.6.1 Current Implementation
+
+The chaining mechanism can be used by the various commands available The following are the types of command that can be chained:
+- list
+- add
+- edit
+
+The following sequence diagram shows how the chaining works after command is entered:
+
+![ChainCommandFeature](diagrams/ChainCommandFeature.jpg)
+
+*Figure 10. Sequence diagram of chaining feature*
+
+<br>
+
+#### 3.6.2 Design Considerations
+
+Aspect: Which features to chain
+
+>Alternative 1 (current choice): Allow only certain commands to be chained.
+>* Pros: Able to guarantee that no abnormal behaviour will happen.
+>* Cons: User must know which commands can be chained.
+
+>Alternative 2: Allow all commands to be chained.
+>* Pros: Easy to implement.
+>* Cons: May give abnormal behaviour.
+
+<br>
+<br>
+
+### 3.7 Move feature
+This feature allows the user to manually `move` activities from one position to another position 
+
+The following sequence diagram shows how the `move` command is executed, where index1 is the position to be moved from and index 2 is the position to be moved below. 
+
+![Move_command](diagrams/moveCommand.png)
```

Comment:
![image](https://user-images.githubusercontent.com/46754292/97526283-c2b75f00-19e3-11eb-8c52-6065f783f383.png)
Is this arrowhead a typo?